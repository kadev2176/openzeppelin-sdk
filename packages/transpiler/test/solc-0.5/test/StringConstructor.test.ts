import { shouldTranspileToValidContract } from './LocalContract.behaviour';

describe(`StringConstructor contract`, (): void => {
  shouldTranspileToValidContract('StringConstructor', {
    StringConstructor: {
      path: 'StringConstructor',
      fileName: 'StringConstructor',
      contracts: ['StringConstructor'],
    },
  });
});
