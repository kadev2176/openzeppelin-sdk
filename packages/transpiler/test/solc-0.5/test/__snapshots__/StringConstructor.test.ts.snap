// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringConstructor contract is converted to a valid StringConstructorUpgradeable contract 1`] = `
"pragma solidity ^0.5.0;
import \\"@openzeppelin/upgrades/contracts/Initializable.sol\\";

contract StringConstructorUpgradeable is Initializable {
    function initialize(string calldata message) external initializer {
        __init(true, message);
    }

    function __init(bool callChain, string memory message) internal {
        
        
        

    
    }

    
}"
`;
